packages/rs-dapi-client/src/platform/get_documents.rs

```diff
@@ -0,0 +1,186 @@
+//! `GetDocuments*` requests.
+
+use dapi_grpc::platform::v0::{self as platform_proto, Proof, ResponseMetadata};
+use drive::query::{OrderClause, StartClause, WhereClause};
+
+use crate::{platform::IncompleteMessage, transport::TransportRequest, DapiRequest, Settings};
+
+fn to_proto_where_clause(
+    where_clause: WhereClause,
+) -> platform_proto::get_documents_request::WhereClause {
+    let mut value_cbor_bytes = Vec::new();
+    ciborium::into_writer(&where_clause.value, &mut value_cbor_bytes)
+        .expect("byte slice is a safe writer");
+
+    platform_proto::get_documents_request::WhereClause {
+        field: where_clause.field,
+        operator: where_clause.operator as i32,
+        value: value_cbor_bytes,
+    }
+}
+
+fn to_proto_order_clause(
+    order_clause: OrderClause,
+) -> platform_proto::get_documents_request::OrderClause {
+    platform_proto::get_documents_request::OrderClause {
+        field: order_clause.field,
+        ascending: order_clause.ascending,
+    }
+}
+
+fn to_proto_start(start_clause: StartClause) -> platform_proto::get_documents_request::Start {
+    use platform_proto::get_documents_request::Start as ProtoStart;
+    match start_clause {
+        StartClause::StartAt(x) => ProtoStart::StartAt(x.to_vec()),
+        StartClause::StartAfter(x) => ProtoStart::StartAfter(x.to_vec()),
+    }
+}
+
+/// Request documents bytes.
+#[derive(Debug)]
+pub struct GetDocuments {
```

Comment:
Everything except `data_contract_id` and `document_type` are optional?

Reply:
I'm following what protos says, for some reason `limit` is not optional

vecs don't have to be optional as they can be empty

Reply:
@lklimek do you have any idea why limit is not optional because it should be optional and we have default value for it

Reply:
Note that in proto3 the default representation for all user-defined message types is Option<T>, and for scalar types just T (during decoding, a missing value is populated by T::default()). If you need a witness of the presence of a scalar type T, use the optional modifier to enforce an Option<T> representation in the generated Rust struct.